pkgs/servers/sql/postgresql/pg_config.sh

```diff
@@ -0,0 +1,32 @@
+# The real pg_config needs to be in the same path as the "postgres" binary
+# to return proper paths. However, we want it in the -dev output to prevent
+# cyclic references and to prevent blowing up the runtime closure. Thus, we
+# have wrapped -dev/bin/pg_config to fake its argv0 to be in the default
+# output. Unfortunately, pg_config tries to be smart and tries to find itself -
+# which will then fail with:
+#   pg_config: could not find own program executable
+# To counter this, we're creating *this* fake pg_config script and put it into
+# the default output. The real pg_config is happy.
+# Some extensions, e.g. timescaledb, use the reverse logic and look for pg_config
+# in the same path as the "postgres" binary to support multi-version-installs.
+# Thus, they will end up calling this script during build, even though the real
+# pg_config would be available on PATH, provided by nativeBuildInputs. To help
+# this case, we're checking whether *this* script is the same as the one available
+# in PATH, and if not we're redirecting the call there.
+# If this script is called directly, we can't do anything except to error out.
+
+# path of *this* script
+my_path="$(readlink -f -- "$0")"
+# path of pg_config on PATH
+on_path="$(readlink -f -- "$(type -P pg_config)")"
+
+if [[ "$my_path" == "$on_path" ]]; then
```

Comment:
Ehm no that I see it, this doesn't take the case into account where `postgresql` is not in PATH, i.e. when I did `./result/bin/pg_config `locally, I just got a `1` as exit code in line 21, we want a reasonable output here as well I'd say.